tests/mongo-tools/dump_restore_test.go

```diff
@@ -21,57 +21,77 @@ import (
 
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
-	"go.mongodb.org/mongo-driver/bson"
 
 	"github.com/FerretDB/dance/tests/common"
 )
 
 func TestDumpRestore(t *testing.T) {
 	ctx, db := common.Setup(t)
 
-	// TODO restart existing dump from https://github.com/mcampo2/mongodb-sample-databases instead
-	_, err := db.Collection("test").InsertOne(ctx, bson.D{{"foo", "bar"}})
+	localActualPath := filepath.Join("..", "..", "dumps", "actual")
+	containerActualPath := "/dumps/"
+
+	localExpectedPath := filepath.Join("..", "..", "dumps", "expected")
+	containerExpectedPath := "/sample-dumps/"
+
+	dbName := "sample_geospatial"
+	db = db.Client().Database(dbName)
+
+	// cleanup database
+	err := db.Drop(ctx)
 	require.NoError(t, err)
 
-	localRoot := filepath.Join("..", "..", "dumps")
-	containerRoot := "/dumps/"
+	// restore a database from a sample dump
+	err = runDockerComposeCommand(
+		"mongorestore",
+		"--nsInclude", dbName+".*",
+		"--verbose",
+		"--uri", "mongodb://host.docker.internal:27017/",
+		"--noIndexRestore",
+		containerExpectedPath,
+	)
+	require.NoError(t, err)
 
 	expectedState := getDatabaseState(t, ctx, db)
 
 	// pre-create directories to avoid permission issues
-	err = os.Chmod(localRoot, 0o777)
+	err = os.Chmod(localActualPath, 0o777)
 	require.NoError(t, err)
-	err = os.RemoveAll(filepath.Join(localRoot, db.Name()))
+	err = os.RemoveAll(filepath.Join(localActualPath, dbName))
 	require.NoError(t, err)
-	err = os.Mkdir(filepath.Join(localRoot, db.Name()), 0o777) // 0o777 is typically downgraded to 0o755 by umask
+	err = os.Mkdir(filepath.Join(localActualPath, dbName), 0o777) // 0o777 is typically downgraded to 0o755 by umask
 	require.NoError(t, err)
-	err = os.Chmod(filepath.Join(localRoot, db.Name()), 0o777) // fix after umask
+	err = os.Chmod(filepath.Join(localActualPath, dbName), 0o777) // fix after umask
 	require.NoError(t, err)
 
 	// dump a database
 	err = runDockerComposeCommand(
 		"mongodump",
-		"--out", containerRoot,
-		"--db", db.Name(),
+		"--out", containerActualPath,
+		"--db", dbName,
 		"--verbose",
 		"mongodb://host.docker.internal:27017/",
 	)
 	require.NoError(t, err)
 
 	// cleanup database
-	ctx, db = common.Setup(t)
+	err = db.Drop(ctx)
+	require.NoError(t, err)
 
 	// restore a database based on created dump
 	err = runDockerComposeCommand(
 		"mongorestore",
-		"--dir", filepath.Join(containerRoot, db.Name()),
-		"--db", db.Name(),
+		"--nsInclude", dbName+".*",
 		"--verbose",
 		"mongodb://host.docker.internal:27017/",
+		containerActualPath,
 	)
 	require.NoError(t, err)
 
-	// get database state after restore
+	// get database state after restore and compare it
 	actualState := getDatabaseState(t, ctx, db)
 	assert.Equal(t, expectedState, actualState)
+
+	// compare dump files
+	compareDirs(t, filepath.Join(localExpectedPath, dbName), filepath.Join(localActualPath, dbName), "/*.metadata.json")
```

Comment:
Let's maybe add a comment why we ignore metadata.json files.

Reply:
done!